playground/miden-wasm/src/utils_input.rs

```diff
@@ -14,11 +16,195 @@ pub struct Outputs {
     pub proof: Option<Vec<u8>>,
 }
 
-/// The InputFile is a struct needed for Miden and not yet exposed.
-#[derive(serde::Deserialize, serde::Serialize)]
+// MERKLE DATA
+// ================================================================================================
+
+/// Struct used to deserialize merkle data from input file. Merkle data can be represented as a
+/// merkle tree or a sparse merkle tree.
+#[derive(serde::Deserialize)]
+pub enum MerkleData {
+    /// String representation of a merkle tree.  The merkle tree is represented as a vector of
+    /// 32 byte hex strings where each string represents a leaf in the tree.
+    #[serde(rename = "merkle_tree")]
+    MerkleTree(Vec<String>),
+    /// String representation of a sparse merkle tree. The sparse merkle tree is represented as a
+    /// vector of tuples where each tuple consists of a u64 node index and a 32 byte hex string
+    /// representing the value of the node.
+    #[serde(rename = "sparse_merkle_tree")]
+    SparseMerkleTree(Vec<(u64, String)>),
+}
```

Comment:
nit: I would use "Merkle" instead of "merkle" everywhere

Reply:
I'm not sure about this.  I think it would look inconsistent when defining the `.inputs` file.  Here's an example of how it would look.  Happy to change it if you prefer it.

```json
{
    "operand_stack": ["1"],
    "advice_stack": ["1", "2", "3", "4"],
    "advice_map": {
        "0000000000000000000000000000000000000000000000000000000000000000": [5, 6, 7, 8],
        "0000001000000000000000000000000000000000000000000000000000000000": [9, 10, 11, 12]
    },
    "merkle_store": [
        {"MerkleTree": [
            "0000000000000000000000000000000000000000000000000000000000000000", 
            "0000000000000000000000000000000000000000000000000000000000000000", 
            "0000000000000000000000000000000000000000000000000000000000000000", 
            "0000000000000000000000000000000000000000000000000000000000000000"]},
        {"SparseMerkleTree": [
            [1, "0000000000000000000000000000000333000000000000000000000000000000"],
            [3, "0000000000000000000000000000000222000000000000000000000000000000"]
        ]}
    ]
}
```

Reply:
Not sure if you were referring to the comment or the serde renaming.

Reply:
Oh yeah, I just meant in the comments. It was a very nitty nit :)